homeassistant/components/renault/sensor.py

```diff
@@ -189,17 +188,22 @@ def _get_utc_value(entity: RenaultSensor[T]) -> datetime:
         state_class=SensorStateClass.MEASUREMENT,
     ),
     RenaultSensorEntityDescription(
+        # For vehicles that DO NOT report charging power in watts, this seems to
+        # correspond to the maximum power that would be admissible by the car based
+        # on the battery state, regardless of the type of charger.
         key="charging_power",
         condition_lambda=lambda a: not a.details.reports_charging_power_in_watts(),
         coordinator="battery",
         data_key="chargingInstantaneousPower",
-        device_class=SensorDeviceClass.CURRENT,
+        device_class=SensorDeviceClass.POWER,
         entity_class=RenaultSensor[KamereonVehicleBatteryStatusData],
-        name="Charging power",
-        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        name="Admissible charging power",
+        native_unit_of_measurement=POWER_KILO_WATT,
         state_class=SensorStateClass.MEASUREMENT,
     ),
     RenaultSensorEntityDescription(
+        # For vehicles that DO report charging power in watts, this is the power
+        # effectively being transferred to the car.
         key="charging_power",
```

Comment:
Are these two entity descriptions ("charging_power") exclusive? Otherwise it looks like there could be a conflict for the unique_id if there are two entities with the same key for the same vehicle.

Reply:
Yes they are exclusive, based on `condition_lambda` just underneath.
The unique_id is the same because they come from the same data key on the Renault API.